File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
long long t=1;
int g=0;
for(int i:nums)
{t*=i==0?1:i;
g+=i==0?1:0;
}
if(g>1)                // more than 1 zeros wala case
{
    vector<int> v(nums.size(),0);
    return v;
}
for(int i=0;i<nums.size();i++)
{
    if(g==1)
    nums[i]=nums[i]==0?t:0;
    else
nums[i]=nums[i]==0?t:t/nums[i];
}
return nums;
}
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n=nums.length;
        int left[]=nums.clone();
        left[0]=nums[0];
        int right[]=nums.clone();
        right[n-1]=nums[n-1];
        for(int i=1;i<n;i++)
        {
            left[i]*=left[i-1];
        }
        for(int i=n-2;i>=0;i--)
        {
            right[i]*=right[i+1];
        }
        for(int i=0;i<n;i++)
        {
            int prod=1;
            if(i>0)
            prod=left[i-1];
            if(i<n-1)
            prod*=right[i+1];
            nums[i]=prod;
        }
        return nums;
    }
}